server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb

```diff
@@ -0,0 +1,212 @@
+##########################GO-LICENSE-START################################
+# Copyright 2014 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################GO-LICENSE-END##################################
+
+module Admin::Materials
+  class PluggableScmController < AdminController
+    include PipelineConfigLoader
+
+    load_pipeline_for_all_actions
+
+    def show_existing
+      assert_load :material, PluggableSCMMaterialConfig.new
+      assert_load :scms, @cruise_config.getSCMs()
+      render layout: false
+    end
+
+    def choose_existing
+      assert_load :material, PluggableSCMMaterialConfig.new
+
+      create_failure_handler = proc do |result, all_errors|
+        @errors = flatten_all_errors(all_errors)
+        @scms = @cruise_config.getSCMs()
+        render :template => "/admin/materials/pluggable_scm/show_existing", :status => result.httpCode(), :layout => false
+      end
+
+      save_popup(params[:config_md5], get_choose_existing_command, create_failure_handler, {:controller => '/admin/materials', :stage_parent => "pipelines", :current_tab => params[:current_tab]}) do
```

Comment:
Line is too long. [198/80]<br>Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.